### PR TITLE
Read value ref object

### PR DESCRIPTION
**Pull request by gleonoliva**
Implementation of function readValueRefObject.

This function would be used for getting all different fields and valueRefs in a Cluster or Cluster-like structures (AnalogWaveform, Complex, Timestamp)

### CODE REVIEW

File: source/include/CEntryPoints.h [hunk 1]
```diff
@@ -76,6 +76,7 @@ VIREO_EXPORT Boolean TypeRef_HasCustomDefault(TypeRef typeRef);
 VIREO_EXPORT EncodingEnum TypeRef_BitEncoding(TypeRef typeRef);
 VIREO_EXPORT Int32 TypeRef_Alignment(TypeRef typeRef);
 VIREO_EXPORT const char* TypeRef_Name(TypeManagerRef typeManager, TypeRef typeRef);
+VIREO_EXPORT const char* TypeRef_ElementName(TypeManagerRef typeManager, TypeRef typeRef);
```
Comment (gleonoliva) [inline source/include/CEntryPoints.h]: I wonder if we should change the signature of these to return an `EggShellResult` and stop using `cwrap`ing them.

Comment (rajsite) [reply source/include/CEntryPoints.h]: We can avoid cwrapping them after we enable reading SubString support so we aren't quite there yet

File: source/core/module_typeHelpers.js [hunk 2]
```diff
@@ -287,6 +293,8 @@
                 dispatcher: dispatchVisitArray
             },
             {
+                // Cluster is evaluated last, since internally Complex, AnalogWaveform and Timestamps
```
Comment (rajsite) [inline source/core/module_typeHelpers.js]: We should add comments for the other types above where order is significant like checking String before Array, etc

File: test-it/karma/publicapi/ReadValueRefObject.Test.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        it('returns an object with keys "real" and "imaginary" and valueRefs as values', function () {
+            var objectValueRef = vireo.eggShell.readValueRefObject(valueRef);
+            expectValidValueRef(objectValueRef.real);
+            expectValidValueRef(objectValueRef.imaginary);
+        });
+    });
+
+    it('throws for any other type', function () {
+        var tryReadValueRefObject = function (path) {
```
Comment (rajsite) [inline test-it/karma/publicapi/ReadValueRefObject.Test.js]: nice helper 👍

**Pull request merged**